### ISSUE TITLE
Ability to query by `release.stage` in Discover

### ISSUE THREAD

**Issue by arifken** [getsentry/sentry#70690]
### Problem Statement

For our Android app, our CD system distributes builds for every `main` and MR branch to employees and other folks within the organization that are doing testing. These releases are internal and do not appear as `release.stage:adopted`. We'd like to be able to query these errors (coming from employees) separately from errors coming from end users (`release.stage:adopted`). Due to our process of promoting internal track builds to production, we don't want to use a separate versioning scheme for employee builds. 

### Solution Brainstorm

Add the ability in Discover (and in dashboards) to query from the Error data source by `release.stage:adopted`. Currently you can only do this on the "Releases" data source, but for Errors/events you can only do `release:<specific version`

### Product Area

Discover

**Comment by gggritso** [comment]
@arifken thanks for raising this! It seems like [tags](https://docs.sentry.io/platforms/android/enriching-events/tags/) would be a good solution here, does that work? If not, could you explain more what you're trying to achieve?

**Comment by arifken-zg** [comment]
Hey sorry for the delayed response. The reason tags don't work for this is that we want to query for a range of dynamic values, not a single one. the challenge is that "adopted" releases is a dynamic value comprised of all versions that are receiving a certain amount of traffic (I presume). So it will change over time and is not something we can define in a tag

**Comment by gggritso** [comment]
@arifken-zg gotcha, thanks for clarifying. I took a quick look, and `release.stage` _is_ supported in Discover for the error dataset. Are you seeing an error?

https://docs.sentry.io/concepts/search/searchable-properties/issues/#releasestage

**Comment by arifken** [comment]
Oh, that's awesome -- that must be new (or I guess I missed it!). I just tested it and there's still a bug when using this in Dashboards. When I try to add `release.stage:adopted error.handled:false` in a new "Errors and Transactions" dashboard widget, it gives me an error about selecting a single environment. but the environment selector is grayed out. I also tried adding a widget from Discover, but I get the same error. When I try to save the widget anyway, I get a 400 in the js console in my browser
